src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/InstanceNameTest/InstanceNameTest.cs

```diff
@@ -84,138 +86,135 @@ public static void ConnectManagedWithInstanceNameTest(bool useMultiSubnetFailove
 
-            // Get the instance name
             PropertyInfo instanceNameInfo = dataSrcInfo.GetType().GetProperty("InstanceName", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
             string instanceName = instanceNameInfo.GetValue(dataSrcInfo, null).ToString();
 
-            // Get the port number using the GetPortByInstanceName method of SSRP
-            object port = getPortByInstanceName.Invoke(ssrp, parameters: new object[] { serverName, instanceName, timeoutTimer, false, 0 });
+            object port = getPortByInstanceNameInfo.Invoke(ssrpObj, parameters: new object[] { serverName, instanceName, timeoutTimerObj, false, 0 });
 
-            // Set the resolved port property of datasource
             PropertyInfo resolvedPortInfo = dataSrcInfo.GetType().GetProperty("ResolvedPort", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
             resolvedPortInfo.SetValue(dataSrcInfo, (int)port, null);
 
-            // Prepare the GetSqlServerSPNs method
             string serverSPN = "";
-            MethodInfo getSqlServerSPNs = sniProxy.GetType().GetMethod("GetSqlServerSPNs", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, null, CallingConventions.Any, getSqlServerSPNsTypesArray, null);
+            MethodInfo getSqlServerSPNs = sniProxyObj.GetType().GetMethod("GetSqlServerSPNs", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic, null, CallingConventions.Any, getSqlServerSPNsTypesArray, null);
 
-            // Finally call GetSqlServerSPNs
-            byte[][] result = (byte[][])getSqlServerSPNs.Invoke(sniProxy, new object[] { dataSrcInfo, serverSPN });
+            byte[][] result = (byte[][])getSqlServerSPNs.Invoke(sniProxyObj, new object[] { dataSrcInfo, serverSPN });
 
-            // Example result: MSSQLSvc/machine.domain.tld:port"
             string spnInfo = Encoding.Unicode.GetString(result[0]);
 
-            out_port = (int)port;
-
             return spnInfo;
         }
 
-        private static bool IsKerberos()
+        private static bool IsSPNPortNumberTestForTCP()
         {
-            return (DataTestUtility.AreConnStringsSetup() 
-                 && DataTestUtility.IsNotLocalhost() 
-                 && DataTestUtility.IsKerberosTest 
-                 && DataTestUtility.IsNotAzureServer() 
+            return (IsInstanceNameValid(DataTestUtility.TCPConnectionString)
+                 && DataTestUtility.IsUsingManagedSNI()
+                 && DataTestUtility.IsNotAzureServer()
                  && DataTestUtility.IsNotAzureSynapse());
         }
 
+        private static bool IsInstanceNameValid(string connectionString)
+        {
+            string instanceName = "";
+
+            SqlConnectionStringBuilder builder = new(connectionString);
+
+            DataTestUtility.ParseDataSource(builder.DataSource, out _, out _, out instanceName);
```

Comment:
This doesn't look right to me. ParseDataSource returns a boolean value to determine whether DataSource is valid (or basically some component of the string is formatted correctly). You are only testing for the existance of "InstanceName" (also potentially some kind of valid looking string). Is this a valid use case? "\\\\myTestInstance, 1234"? I think the result will be 'DataSource = false', but you will still have an instanceName = "myTestInstance". Please confirm. Technically you should test both. Datasource = true / InstanceName is valid string

Reply:
The previous test AreConnStringsSetup were just testing for basically non empty strings (of whatever). Since you are using parse dataSource now, which performs greater validation, I'd recommend you do both. Check for DataSource/InstanceName.

Reply:
IsInstanceNameValid now also checks if DataSource is valid.